astroid/nodes/node_ng.py

```diff
@@ -248,15 +263,40 @@ def parent_of(self, node):
                 return True
         return False
 
-    def statement(self):
+    @overload
+    def statement(
+        self, future: Literal[None] = ...
+    ) -> Union["nodes.Statement", "nodes.Module"]:
+        ...
+
+    @overload
+    def statement(self, future: Literal[True]) -> "nodes.Statement":
+        ...
+
+    def statement(
+        self, future: Literal[None, True] = None
+    ) -> Union["nodes.Statement", "nodes.Module", NoReturn]:
```

Comment:
```suggestion
    def statement(
        self, *, future: Literal[None, True] = None
    ) -> Union["nodes.Statement", "nodes.Module", NoReturn]:
```

I should have brought that up earlier but I somehow missed it. What do you think about making `future` a keyword only arg? That will make it explicit, prevent issue if we decide to add additional arguments, and make it easier to deprecate / remove.